packages/react-mutation-mapper/src/component/track/ExonNumTrack.tsx

```diff
@@ -0,0 +1,160 @@
+                        if (
+                            exonStart <= currentUtr.start &&
+                            exonEnd >= currentUtr.start
+                        ) {
+                            weDidIt = true;
+                            let aaLength =
+                                (currentUtr.start -
+                                    exonStart +
+                                    (exonEnd - currentUtr.end)) /
+                                3;
+                            if (aaLength !== 0) {
+                                exonLocList.push([exonArr[i].rank, aaLength]);
+                                totLength += aaLength;
+                            }
+                            break;
+                        }
+                    }
+                    // if there are no utr start sites within exon
+                    if (!weDidIt) {
+                        let aaLength = (exonEnd - exonStart + 1) / 3;
+                        exonLocList.push([exonArr[i].rank, aaLength]);
+                        totLength += aaLength;
+                    }
+                }
+                exonLocList.sort((n1, n2) => n1[0] - n2[0]);
+                if (totLength !== this.props.proteinLength) {
+                    exonLocList[exonLocList.length - 1][1] =
+                        exonLocList[exonLocList.length - 1][1] - 1;
+                    totLength--;
+                }
+                let startOfExon = 0;
+                for (let i = 0; i < exonLocList.length; i++) {
+                    let exonLen = exonLocList[i][1];
+                    data.push([exonLocList[i][0], exonLen, startOfExon]);
+                    startOfExon += exonLen;
+                }
+            }
+        }
+
+        const altColors = ['#007FFF', '#35BAF6'];
+
+        return data.map((exon: number[], index: number) => {
+            let startCodon = exon[2];
+            let endCodon = exon[2] + exon[1];
+            let exonLength = exon[1];
+            let isSkippable = Number.isInteger(exonLength).toString();
+
+            return {
+                color: altColors[index % 2],
+                startCodon: startCodon,
+                endCodon: endCodon,
+                label: exon[0].toString(),
+                labelColor: '#FFFFFF',
+                itemType: TrackItemType.RECTANGLE,
+                tooltip: (
+                    <span>
+                        <h1> Exon {exon[0]} </h1>
+                        <br></br>
+                        Start: {startCodon.toFixed(2).toString()}
+                        <br></br>
+                        End: {endCodon.toFixed(2).toString()}
+                        <br></br>
+                        Exon length (amino acid):{' '}
+                        {exonLength.toFixed(2).toString()}
+                        <br></br>
+                        Exon is a multiple of 3: {isSkippable}
```

Comment:
Not reviewing this for visuals, but I'd think that this reads a bit better.
```suggestion
                        Exon {isSkipable?"is":"isn't"} a multiple of 3
```